pkg/collector/process_test.go

```diff
@@ -628,14 +634,14 @@ func TestDTLSCollectingProcess(t *testing.T) {
 	assert.NotNil(t, cp.templatesMap[1], "DTLS Collecting Process should receive and store the received template.")
 }
 
-func waitForCollectorReady(t *testing.T, address net.Addr) {
+func waitForCollectorReady(t *testing.T, cp *CollectingProcess) {
 	checkConn := func() (bool, error) {
-		if _, err := net.Dial(address.Network(), address.String()); err != nil {
+		if _, err := net.Dial(cp.GetAddress().Network(), cp.GetAddress().String()); err != nil {
 			return false, err
 		}
 		return true, nil
```

Comment:
Could we close the connection before returning true, so we can check for exact connections in this line:
https://github.com/vmware/go-ipfix/pull/114/files#diff-24e923a024538e6ea8a21097adf7e88b862e14a44c743290238252e0a429d1e7R362 ?

Reply:
if we close the connection, we cannot send data to corresponding connection. Do you mean we can create another function to wait for two clients shown up and close the connection?

Reply:
I was talking about the connection created in L639, which we are ignoring right now.

Reply:
Just curious, should waitForCollectorReady return error if collector is not ready after 5 retries?

Reply:
This still seems to be outstanding. Any follow up comment on this?

Reply:
Sorry for the late reply. @srikartati updated the close connection code, now we can check exact number of clients. @shihhaoli yes it will log error if the collector is not ready, which will cause the tests to fail.